src/main/webapp/app/entities/participation/participation.component.html

```diff
@@ -109,11 +109,16 @@ <h2>
                                 <fa-icon icon="file-powerpoint"></fa-icon>
                                 <span class="d-none d-md-inline" jhiTranslate="artemisApp.participation.removePresentation.label">Remove presentation</span>
                             </button>
-                            <jhi-programming-exercise-instructor-trigger-build-button
-                                *jhiHasAnyAuthority="['ROLE_INSTRUCTOR', 'ROLE_ADMIN']"
-                                [participation]="participation"
-                                [showProgress]="true"
-                            ></jhi-programming-exercise-instructor-trigger-build-button>
+                            <ng-container *ngIf="exercise?.type === PROGRAMMING && exercise.isAtLeastInstructor">
+                                <jhi-programming-exercise-instructor-trigger-build-button
+                                    *ngIf="hasLoadedPendingSubmissions; else triggerLoading"
```

Comment:
The following is missing. Why?
*jhiHasAnyAuthority="['ROLE_INSTRUCTOR', 'ROLE_ADMIN']"

Reply:
The problem is that an element can only have one structural element in Angular, while we would need two here (ngIf & jhiHasAnyAuthority). I decided to streamline this with the rest of the file by the use of `exercise.isAtLeastInstructor`:
https://github.com/ls1intum/Artemis/pull/744/files/4bc23e85088a31f116ad130010eaa3b698400eb8#diff-e63d29dc7649c777d5dab501d6e47c9cR112